packages/admin/admin/src/inlineAlert/InlineAlert.tsx

```diff
@@ -0,0 +1,135 @@
+import { Error, Info, Warning } from "@comet/admin-icons";
+import { type ComponentsOverrides, type Theme, type Typography } from "@mui/material";
+import { useThemeProps } from "@mui/material/styles";
+import { type FunctionComponent, type ReactNode } from "react";
+import { FormattedMessage } from "react-intl";
+
+import { type ThemedComponentBaseProps } from "../helpers/ThemedComponentBaseProps";
+import { ActionsContainer, Description, IconContainer, Root, Title } from "./InlineAlert.sc";
+
+export type InlineAlertClassKey = "root" | "iconContainer" | "title" | "description" | "actionsContainer";
+
+type InlineAlertSeverity = "error" | "warning" | "info";
+
+export type InlineAlertProps = ThemedComponentBaseProps<{
+    root: "div";
+    iconContainer: "div";
+    title: typeof Typography;
+    description: typeof Typography;
+    actionsContainer: "div";
+}> & {
+    /**
+     * A description message providing more details about the error.
+     */
+    description?: ReactNode;
+
+    /**
+     * A mapping of descriptions for each error variant.
+     */
+    descriptionMapping?: Record<InlineAlertSeverity, ReactNode>;
+
+    /**
+     * Custom icon to be displayed next to the error message.
+     * If not provided, a default icon based on the variant will be used.
+     */
+    icon?: ReactNode;
+
+    /**
+     * A mapping of icons for each error variant.
+     */
+    iconMapping?: Record<InlineAlertSeverity, ReactNode>;
+
+    /**
+     * Defines the type of error message displayed.
+     * Variants include "error", "warning", and "info".
+     *
+     * @defaultValue "error"
+     */
+    severity?: InlineAlertSeverity;
+
+    /**
+     * Action elements such as buttons or links that allow users to take corrective action.
+     * The actions are shown below the error message.
+     */
+    actions?: ReactNode;
+
+    /**
+     * The title of the error message.
+     */
+    title?: ReactNode;
+
+    /**
+     * A mapping of titles for each error variant.
+     */
+    titleMapping?: Record<InlineAlertSeverity, ReactNode>;
+};
+
+/**
+ * The InlineAlert component is used to display inline error messages within an application.
+ * It provides visual feedback for errors, warnings, or informational messages with optional actions.
+ */
+export const InlineAlert: FunctionComponent<InlineAlertProps> = (inProps) => {
+    const {
+        icon,
+        iconMapping = {
+            error: <Error sx={{ fontSize: "32px" }} color="error" />,
+            warning: <Warning sx={{ fontSize: "32px" }} color="warning" />,
+            info: <Info sx={{ fontSize: "32px" }} color="info" />,
+        },
```

Comment:
Setting an object with all icons in the default value will cause an issue:

When doing the following, the icons `error` and `warning` will not exist, as the whole `iconMapping` object is replaced. Only the info icon should be replaced though. 

```tsx
<InlineAlert iconMapping={{ info: <CustomInfoIcon /> }} />
```

This should not be an issue if you define the default icons like in the docs example: [Overridable icons](https://docs.comet-dxp.com/docs/comet-core-development/creating-customizable-admin-components#overridable-icons)

The same issue applies to `titleMapping` and `descriptionMapping`.

Reply:
Fixed in 89583ea65252e72354d134c95e6ef4d3ff903863